Project/src/main/java/app/models/User.java

```diff
@@ -15,19 +15,34 @@
 
     private String password;
 
-    private String role;
+    @OneToOne
+    private Role role;
 
-    @ManyToMany(mappedBy = "user", cascade = CascadeType.ALL)
-    private List<File> files;
+    @ManyToMany(mappedBy = "users", cascade = CascadeType.ALL)
+    private List<Article> authorizedArticles;
 
     public User() {
         this(null, null, null);
     }
+
     public User(String username, String password, String role){
         this.username = username;
         this.password = password;
-        this.role = role;
-        files = new ArrayList<>();
+        authorizedArticles = new ArrayList<>();
+
+        if (role != null) {
+            switch (role) {
+                case Role.reviewer: {
+                    this.role = new Reviewer();
+                }
+                case Role.editor: {
+                    this.role = new Editor();
+                }
+                case Role.submitter: {
+                    this.role = new Submitter();
+                }
+            }
+        }
```

Comment:
This is poor design.
a) You have a factory process in a constructor
b) There is no need for a user to have a role like this, their role can/should be checked for action purposes
c) Maybe try making editor/submitter/ an extension of a user